include/operators/Not.h

```diff
@@ -31,13 +31,27 @@ namespace dnnc {
 template <typename T> class Not : public baseOperator<T> {
   //  Not attributes
 public:
+  
   Not(std::string name = "opNot") : baseOperator<T>(opNot, name) {}
 
-  // bool getAttribute<int>(OPATTR attrName, int& obj) ;
+  /*! Element wise Negate-Function*/
+  static T negate_function(T x) {
+    return (!x);
+  }
+  tensor<T> compute(tensor<T> &a /*!<[bool]: ND tensor*/) {
+
+    if (!(this->template type_check<bool>()))
+      throw std::invalid_argument(
+        "Constrain input and output types to bool tensors.");
+
+    tensor<T> result(a.shape(), a.name());
+    
+    DNNC_EIGEN_ARRAY_MAP(eigenVector, a);
+    DNNC_EIGEN_VECTOR_CTOR(T) eResult;
+    eResult.array() = eigenVector.array().unaryExpr(&negate_function);
```

Comment:
will NOT EIGEN operator (!) be more appropriate here. Example: http://eigen.tuxfamily.org/dox/classEigen_1_1ArrayBase.html#a982684b90fd894b31b3a6722a5d994a5